generator/src/FluentStageFactoryGenerator.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\CodeGenerator;
+
+use MongoDB\BSON\Decimal128;
+use MongoDB\BSON\Document;
+use MongoDB\BSON\Int64;
+use MongoDB\BSON\PackedArray;
+use MongoDB\BSON\Serializable;
+use MongoDB\BSON\Timestamp;
+use MongoDB\BSON\Type;
+use MongoDB\Builder\Expression\ArrayFieldPath;
+use MongoDB\Builder\Expression\FieldPath;
+use MongoDB\Builder\Expression\ResolvesToArray;
+use MongoDB\Builder\Expression\ResolvesToObject;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Builder\Type\AccumulatorInterface;
+use MongoDB\Builder\Type\ExpressionInterface;
+use MongoDB\Builder\Type\Optional;
+use MongoDB\Builder\Type\QueryInterface;
+use MongoDB\Builder\Type\Sort;
+use MongoDB\CodeGenerator\Definition\GeneratorDefinition;
+use MongoDB\Model\BSONArray;
+use Nette\PhpGenerator\ClassType;
+use Nette\PhpGenerator\Method;
+use Nette\PhpGenerator\PhpNamespace;
+use Nette\PhpGenerator\TraitType;
+use RuntimeException;
+use stdClass;
+use Throwable;
+
+use function array_key_last;
+use function array_map;
+use function assert;
+use function implode;
+use function sprintf;
+
+/**
+ * Generates a fluent factory class for aggregation pipeline stages.
+ * The method definition is based on the manually edited static class
+ * that imports the stage factory trait.
+ */
+class FluentStageFactoryGenerator extends OperatorGenerator
+{
+    /**
+     * All public of this class are duplicated as instance methods of the
+     * fluent factory class.
+     */
+    private const FACTORY_CLASS = Stage::class;
+
+    public function generate(GeneratorDefinition $definition): void
+    {
+        $this->writeFile($this->createFluentFactoryClass($definition));
+    }
+
+    private function createFluentFactoryClass(GeneratorDefinition $definition): PhpNamespace
+    {
+        $namespace = new PhpNamespace($definition->namespace);
+        $class = $namespace->addClass('FluentFactory');
+
+        $namespace->addUse(Pipeline::class);
+        $namespace->addUse(Decimal128::class);
+        $namespace->addUse(Document::class);
+        $namespace->addUse(Int64::class);
+        $namespace->addUse(PackedArray::class);
+        $namespace->addUse(Serializable::class);
+        $namespace->addUse(Timestamp::class);
+        $namespace->addUse(Type::class);
+        $namespace->addUse(ArrayFieldPath::class);
+        $namespace->addUse(FieldPath::class);
+        $namespace->addUse(ResolvesToArray::class);
+        $namespace->addUse(ResolvesToObject::class);
+        $namespace->addUse(Pipeline::class);
+        $namespace->addUse(AccumulatorInterface::class);
```

Comment:
Does the generated code include fully qualified names without these use statements, or would the code break? IMO, in generated code it's fine to not rely on `use` statements, alternatively we could always generate the code with fully qualified names and then let phpcbf fix whatever it can (without actually checking the code against our styleguide so that manual fixes need not to be applied).

Reply:
This breaks phpdoc.